File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/1943-describe-the-painting.js
var splitPainting = function(segments) {
    const events = [];
    for (const [start, end, color] of segments) {
        events.push([start, 'start', color]);
        events.push([end, 'end', color]);
    }
    events.sort((a, b) => a[0] - b[0] || (a[1] === 'start' ? -1 : 1));

    const result = [];
    let prevPos = null;
    let currentColor = 0;
    const activeColors = new Set();

    for (const [pos, type, color] of events) {
        if (prevPos !== null && pos !== prevPos && activeColors.size > 0) {
            if (result.length > 0 && result[result.length - 1][1] === prevPos) {
                const last = result.pop();
                currentColor = last[2] + currentColor;
                result.push([last[0], pos, currentColor]);
            } else {
                result.push([prevPos, pos, currentColor]);
            }
        }

        if (type === 'start') {
            activeColors.add(color);
            currentColor += color;
        } else {
            activeColors.delete(color);
            currentColor -= color;
        }

        prevPos = pos;
    }

    return result;
};

File: leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/1943-describe-the-painting.erl
-module(solution).
-export([split_painting/1]).

split_painting(Segments) ->
    Events = lists:foldl(fun([Start, End, Color], Acc) ->
        [{Start, Color, start} | [{End, Color, end} | Acc]]
    end, [], Segments),
    SortedEvents = lists:keysort(1, Events),
    process_events(SortedEvents, 0, 0, [], []).

process_events([], _PrevPos, _CurrentColor, _CurrentList, Result) ->
    lists:reverse(Result);
process_events([{Pos, Color, Type} | Rest], PrevPos, CurrentColor, CurrentList, Result) ->
    case PrevPos < Pos andalso CurrentColor =/= 0 of
        true ->
            NewResult = [[PrevPos, Pos, CurrentColor] | Result],
            NewCurrentList = [];
        false ->
            NewResult = Result,
            NewCurrentList = CurrentList
    end,
    NewCurrentList1 = case Type of
        start -> [Color | NewCurrentList];
        end -> lists:delete(Color, NewCurrentList)
    end,
    NewColor = case NewCurrentList1 of
        [] -> 0;
        _ -> lists:sum(NewCurrentList1)
    end,
    process_events(Rest, Pos, NewColor, NewCurrentList1, NewResult).